src/fmu/dataio/providers/objectdata/_xtgeo.py

```diff
@@ -32,6 +34,39 @@
 logger: Final = null_logger(__name__)
 
 
+def lack_of_geometry_warn() -> None:
+    warnings.warn(
+        dedent(
+            """
+            From fmu.dataio version 2.3:
+
+            When exporting a grid property, linking it to a geometry is strongly
+            recommended and may be mandatory in the near future!
+            Example:
+
+                grid = xtgeo.grid_from_roxar(project, "Geogrid")
+                poro = xtgeo.gridproperty_from_roxar(project, "Geogrid", "PORO")
+
+                # first export the grid geometry:
+                grd_edata = dataio.ExportData(
+                    config=rmsglobalconfig, content="depth", name="Geogrid"
+                )
+                grid_output = grd_edata.export(grid)
+
+                # Export of the gridproperty points to output of the exported geometry
+                por_edata = dataio.ExportData(
+                    config=rmsglobalconfig,
+                    content={"property": {"is_discrete": False}},
+                    name="PORO",
+                    geometry=grid_output,    # <<<
+                )
+                output = por_edata.export(poro)
+            """
+        ),
+        FutureWarning,
+    )
+
```

Comment:
Will this warning be too intrusive?

Reply:
I wonder if the example should be added into the docs instead. I am not sure if they'll end up more stale at docs or in a warning message, but it is much more usual to include a link here rather than the code example.

Reply:
Move to docs :+1: